spec/puppet-server-lib/puppet/jvm/compiler_spec.rb

```diff
@@ -48,20 +48,74 @@
 
       before(:each) do
         FileUtils.mkdir_p(File.join(Puppet[:environmentpath], environment))
+      end
 
+      it 'by default uses the classified environment' do
         Puppet::Node.indirection.expects(:find).returns(
           Puppet::Node.new(certname, environment: 'production')
         )
-      end
 
-      it 'by default uses the classified environment' do
         expect(node.environment.name).to eq(:production)
       end
 
+      context 'and facts are not submitted' do
+        let(:facts) { nil }
+
+        it 'requests facts from pdb after classifying and attempts to classify again' do
+          pdb_environments = sequence('pdb environments')
```

Comment:
Oh no, are we still using mocha here? :(

(I assumed we weren't, so it took me ages to figure out what this was doing :P)

Reply:
It looks like we are relying on Puppet for this configuration. Is their mocha removal going to break all our spec tests?

Reply:
Yeah we're getting everything from puppet. I've experimented with not doing so and it's not too hard. I can put up something for that after this.

Reply:
Oh that's good to hear. Thanks for taking a look!